File: app/src/main/java/com/booboot/vndbandroid/activity/URLActivity.java
package com.booboot.vndbandroid.activity;

import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.booboot.vndbandroid.bean.vndb.Links;
import com.booboot.vndbandroid.bean.vndbandroid.Theme;
import com.booboot.vndbandroid.util.SettingsManager;
import com.booboot.vndbandroid.util.Utils;

public class URLActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTheme(Theme.THEMES.get(SettingsManager.getTheme(this)).getStyle());

        Uri data = getIntent().getData();
        String url = data.toString().substring(getPackageName().length() + 3);
        if (url.startsWith("/")) url = Links.VNDB + url;
        Utils.openURL(this, url);
        finish();
    }
}

File: app/src/main/java/com/booboot/vndbandroid/ui/url/URLActivity.kt
package com.booboot.vndbandroid.ui.url

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.booboot.vndbandroid.extensions.openURL

class URLActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        intent.data?.toString()?.substring(packageName.length + 3)?.let {
            openURL(it)
        }
        finish()
    }
}